packages/core/src/hooks/useSelect/index.ts

```diff
@@ -35,15 +35,24 @@ export type UseSelectProps<TQueryFnData, TError, TData> = {
    */
   resource: string;
   /**
-   * Set the option's value
+   * Set the option's label value
    * @default `"title"`
    */
-  optionLabel?: keyof TData extends string ? keyof TData : never;
+  optionLabel?: keyof TData extends string
+    ? keyof TData
+    : never | ((item: TData) => string);
   /**
-   * Set the option's label value
+   * Set the option's value
    * @default `"id"`
    */
-  optionValue?: keyof TData extends string ? keyof TData : never;
+  optionValue?: keyof TData extends string
+    ? keyof TData
+    : never | ((item: TData) => string);
+  /**
+   * Set search label
+   * @default `"title"``
+   */
+  searchLabel?: keyof TData extends string ? keyof TData : never;
```

Comment:
I don't think naming and JSDoc is descriptive here
Maybe It can be something like this

```suggestion
  /**
   * Field name to search for
   * @default `"title"``
   */
  fieldToSearch?: keyof TData extends string ? keyof TData : never;
```

Reply:
Nice catch. Maybe `searchField` ?